cloudsync_gdrive.py

```diff
@@ -312,56 +326,103 @@ def events(self) -> Generator[Event, None, None]:  # pylint: disable=too-many-lo
-                if fil:
-                    if fil.get('mimeType') == self._folder_mime_type:
-                        otype = DIRECTORY
-                    else:
-                        otype = FILE
-                else:
-                    otype = NOTKNOWN
-
-                ohash = None
-                path = self._path_oid(oid, use_cache=False)
+            if new_cursor and page_token and new_cursor != page_token:
+                self.__cursor = new_cursor
+            page_token = response.get('nextPageToken')
 
-                event = Event(otype, oid, path, ohash, exists, ts, new_cursor=new_cursor)
+    def _convert_to_event(self, change, new_cursor):
+        ts = arrow.get(change.get('time')).float_timestamp
+        oid = change.get('fileId')
+        exists = None
+        if 'removed' in change.keys():
+            exists = not change.get('removed')
+
+        fil = change.get('file')
+        if fil:
+            if fil.get('mimeType') == self._folder_mime_type:
+                otype = DIRECTORY
+            else:
+                otype = FILE
+        else:
+            otype = NOTKNOWN
 
-                remove = []
-                for cpath, coid in self._ids.items():
-                    if coid == oid:
-                        if cpath != path:
-                            remove.append(cpath)  # remove the event's item if it's path changed
+        ohash = None
+        path = self._path_oid(oid, use_cache=False)
+        if not path:
+            # Trashed
+            exists = False
 
-                    if path and otype == DIRECTORY and self.is_subpath(path, cpath):
-                        remove.append(cpath)  # if the event's item is a folder, uncache its children
+        event = Event(otype, oid, path, ohash, exists, ts, new_cursor=new_cursor)
 
-                for r in remove:
-                    self._ids.pop(r, None)
+        remove = []
+        for cpath, coid in self._ids.items():
+            if coid == oid:
+                if cpath != path:
+                    remove.append(cpath)  # remove the event's item if it's path changed
 
-                if path:
-                    self._ids[path] = oid
+            if path and otype == DIRECTORY and self.is_subpath(path, cpath):
+                remove.append(cpath)  # if the event's item is a folder, uncache its children
 
-                log.debug("converted event %s as %s", change, event)
+        for r in remove:
+            self._ids.pop(r, None)
 
-                yield event
+        if path:
+            self._ids[path] = oid
+
+        return event
+
+    def _filter_event(self, event):
+        # event filtering based on root path and event path
+
+        if not event:
+            return EventFilter.IGNORE
+
+        if not self._root_path:
+            return EventFilter.PROCESS
+
+        state_path = self.sync_state.get_path(event.oid)
+        prior_subpath = self.is_subpath_of_root(state_path)
+        if not event.exists:
```

Comment:
curious - are gdrive delete events the same as onedrive in that they don't have a path?

Reply:
The current implementation populates a path but we are moving away from that. Path will be `None` for all events going forward to cut down on duplicate api hits